internal/generate/generate.go

```diff
@@ -208,10 +208,45 @@ func GeneratePkgs(directory string, goMod2NixPath string, numWorkers int) ([]*sc
 		return nil, err
 	}
 
+	// Dependencies are downloaded, run `go mod vendor` to obtain `vendor/modules.txt` without reverse engineering
+	const tmpVendorEnvRelative = "vendor-gomod2nix"
+	tmpVendorEnvAbsolute := filepath.Join(directory, tmpVendorEnvRelative)
+	err = os.RemoveAll(tmpVendorEnvAbsolute)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = os.RemoveAll(tmpVendorEnvAbsolute)
+	}()
+
+	var modulesTxt string
+	{
+		log.Info("Obtaining modules.txt")
+		cmd := exec.Command(
+			"go", "mod", "vendor", "-o", tmpVendorEnvRelative,
```

Comment:
I haven't tested this, but I believe that one would need to execute `go work vendor` instead to mark all submodules' explicit requires as such.

There are other issues getting submodules to work with gomod2nix, but if this doesn't cause a problem for projects without `go.work` files, it would get submodule support one step closer.